hash.c

```diff
@@ -2240,23 +2159,21 @@ rb_hash_fetch(VALUE hash, VALUE key)
 
 /*
  *  call-seq:
- *    hash.default -> value
- *    hash.default(key) -> value
+ *    hash.default(key = self.default) -> value
```

Comment:
I don't think these two lines can be combined.  The behavior in the 0 argument case when a default proc is used is to always return nil.  The behavior in the 1 argument case when a default proc is used is to call the default proc with the key.  The behavior of the 0 argument case with the default proc should be documented (no need for an example).

Reply:
Fixed.